replica/table.cc

```diff
@@ -2559,8 +2559,16 @@ class compaction_group::table_state : public compaction::table_state {
     const tombstone_gc_state& get_tombstone_gc_state() const noexcept override {
         return _t.get_compaction_manager().get_tombstone_gc_state();
     }
+    compaction_backlog_tracker& get_backlog_tracker() const noexcept override {
+        return _cg.get_backlog_tracker();
+    }
 };
 
+compaction_backlog_tracker& compaction_group::get_backlog_tracker() const noexcept {
+    // FIXME: will return group's own compaction_backlog_tracker once it's decoupled from compaction_strategy.
+    return _t.get_compaction_strategy().get_backlog_tracker();
+}
```

Comment:
Will any table_state have a different implementation?

Reply:
There is table_state::get_compaction_strategy(). So compaction_manager can do this itself.

Reply:
That's for retrieving the backlog owned by a compaction group, not making it.

compaction_strategy now implements make_backlog_tracker(), which is used by compaction group to create a new one when setting or changing the compaction strategy. The compaction group is responsible for managing it.

compaction_group has:

```
compaction_backlog_tracker& compaction_group::get_backlog_tracker() const noexcept {
    return *_backlog_tracker;
}
```

See this patch: https://github.com/scylladb/scylladb/pull/11762/commits/a590e7106f968c94caf965254c91fbbd704fc7c9